### PR TITLE
Update 2013-03-14-using-bigmemory-with-rcpp.md

### PR DESCRIPTION
**Pull request by sritchie73**
Added the BH dependency to the Rcpp header (required to compile successfully), and added a note about package development (you need C++11 as a SystemRequirement to compile without warnings).

### CODE REVIEW

**Comment by eddelbuettel** [comment]
Good points -- thanks for picking up on that.

**Comment by eddelbuettel** [comment]
Oh, but you edited the _wrong file_ -- the .md is generated.  I'll fold it over by hand in a minute.

**Comment by sritchie73** [comment]
Whoops, my apologies!

File: _posts/2013-03-14-using-bigmemory-with-rcpp.md [hunk 1]
```diff
@@ -19,19 +19,25 @@ This example demonstrates how to write functions in `Rcpp` that operate on
 `big.matrix` objects. Here we implement the equivalent of `colSums`:
 
 {% highlight cpp %}
+// Since bigmemory now accesses Boost headers from the BH package,
+// we need to make sure we do so as well in this Rcpp::depends comment 
+// when including both the Rcpp and bigmemory header files.
+
+// [[Rcpp::depends(BH)]]
 #include <Rcpp.h>
```
Comment (eddelbuettel) [inline _posts/2013-03-14-using-bigmemory-with-rcpp.md:27]: We don't need this twice.  BH is also dependended upon on line 30/32 per my last change (a few days ago?)

File: _posts/2013-03-14-using-bigmemory-with-rcpp.md [hunk 2]
```diff
@@ -19,19 +19,25 @@ This example demonstrates how to write functions in `Rcpp` that operate on
...
+// headers -- thanks to the magic of Rcpp attributes.
 // [[Rcpp::depends(BH, bigmemory)]]
 #include <bigmemory/MatrixAccessor.hpp>
 
 #include <numeric>
 
+// You will also need to add the following to your DESCRIPTION file if
+// you're developing a package so that it will compile without warnings:
+// SystemRequirements: C++11
+
```
Comment (eddelbuettel) [inline _posts/2013-03-14-using-bigmemory-with-rcpp.md:40]: That is going a little off the plantation here.  What we should add is the cpp11 plugin.

Comment (sritchie73) [inline _posts/2013-03-14-using-bigmemory-with-rcpp.md:40]: Now that I go back to my code, you're correct about not needing the import
twice. I'm not sure why that solved the problems I was having a few days
ago, but everything compiles fine now I've removed the extraneous import.

I actually tried but couldn't get the Rcpp11 plugin to work. It seemed to
break all of my existing Rcpp code, and I wasn't able to find much in the
way of documentation for Rcpp11 (other than Romain François's abstract for
a tutorial at useR!2014).

Comment (eddelbuettel) [inline _posts/2013-03-14-using-bigmemory-with-rcpp.md:40]: i)  Try updating and see what I just committed. Now builds without warnings (squashed another one about int / unsigned int)

ii)  You are confusing the C++11 language standard (which Rcpp has supported for years, I wrote Rcpp Gallery articles 1 1/2 years ago) with the Rcpp11 project.  This gallery is about Rcpp which allows you to use C++11 or any other C++ feature or dialect without imposing one -- in clear difference to Rcpp11 which imposes one for its internals.

Comment (jjallaire) [inline _posts/2013-03-14-using-bigmemory-with-rcpp.md:40]: Scott,

You can just add:

// [[Rcpp::plugins(cpp11)]]

Which will ensure that the compiler gets the right flag to indicate that
C++ 11 features should be available.

On Fri, Jul 11, 2014 at 9:42 PM, Dirk Eddelbuettel <notifications@github.com

Comment (eddelbuettel) [inline _posts/2013-03-14-using-bigmemory-with-rcpp.md:40]: Yes, see commit  https://github.com/jjallaire/rcpp-gallery/commit/a2c368bb49c1d669f02bbc081ef590ebdee85e41

Comment (sritchie73) [inline _posts/2013-03-14-using-bigmemory-with-rcpp.md:40]: Thanks Dirk and JJ,

I have to admit I know very little C++, so there's been a lot of confusion
for me (although I've used C for comp sci classes in the past).

I still get the warning about long long when using just  //
[[Rcpp::plugins(cpp11)]] , but the warning goes away when adding
SystemRequirements: C++11
In fact having // [[Rcpp::plugins(cpp11)]] doesn't seem to change whether
or not I can compile without warnings. I've created a minimal package
around the Rcpp Gallery post that compiles and builds without warnings and
passes R CMD check (using RStudio and roxygen2 to generate the NAMESPACE
file etc.):

https://github.com/sritchie73/bigmemory-rcpp-test

Comment (eddelbuettel) [inline _posts/2013-03-14-using-bigmemory-with-rcpp.md:40]: Package != Gallery post.  

This is not the place to discuss problem with packages, or how to fix them.

_Please_ use the mailing list for that.

**Comment by eddelbuettel** [comment]
An [updated post is now back at the Gallery](http://gallery.rcpp.org/articles/using-bigmemory-with-rcpp/) but the process of getting there was a tad painful.

May I suggest we first discuss in email next time before sending random pull requests?

Thanks again for the submission. The content was always very nice.

**Pull request merged**